subprojects/execution/src/main/java/org/gradle/internal/execution/steps/Result.java

```diff
@@ -22,7 +22,27 @@
 
 import java.time.Duration;
 
-public interface Result {
+public class Result {
```

Comment:
Unrelated to the PR, but I would consider giving this class a slightly more concrete name like `StepResult`. I think it's good practice to use move descriptive names for public classes, when they are not nested.